### DISCUSSION TITLE
Type Hinting?

### DISCUSSION THREAD

**Discussion by anirudhRowjee** [anirudhRowjee/atlas#7] [Ideas]
I'm considering adding type annotations. What do you guys think?

**Comment by MadhavJivrajani** [comment]
Vvv good idea. Eventhough python is flexible with this, it's a good practice to add type annotations as extensively as possible.

**Reply by anirudhRowjee** [reply]
The initial build I had before I erased the contents of the file and re-started had type hints. I'll add it as an issue

**Reply by roysti10** [reply]
Are you considering mypy checks for it?

**Reply by MadhavJivrajani** [reply]
> The initial build I had before I erased the contents of the file and re-started had type hints. I'll add it as an issue

Just a thought: Maybe making it part of the contribution guidelines (when you have one) would make more sense than it being an issue as it'd be better if that practice is followed while active development rather than someone having to come correct it. So as you mentioned, it's just a single file as of now, maybe it'd be better if you can take care of that in #8 and you could add checks (if they exist) to ensure that its followed in the contributions that are sent in.

**Reply by anirudhRowjee** [reply]
Sure. I wasn't initially consider mypy for it, but It looks to be a good choice. And yeah, it's better that I add it on my own before anything else is done.

**Reply by anirudhRowjee** [reply]
The write is in progress. I'm using type hints for everything.

**Reply by anirudhRowjee** [reply]
Check out the commit referenced here -
https://github.com/anirudhRowjee/atlas/pull/8/commits/1d8301ceee98b5aebdd6a5fd0dea98a781d2ecfd
